src/main/java/org/owasp/validator/html/scan/AntiSamyDOMScanner.java

```diff
@@ -368,8 +372,26 @@ private void actionValidate(int currentStackDepth, Element ele, Node parentNode,
 
         if (processAttributes(ele, tagName, tag, currentStackDepth)) return; // can't process any more if we
 
-        if (isNofollowAnchors && "a".equals(tagNameLowerCase)) {
-            ele.setAttribute("rel", "nofollow");
+        if ("a".equals(tagNameLowerCase)) {
+            boolean addNofollow = isNofollowAnchors;
+            boolean addNoopenerAndNoreferrer = false;
+
+            if (isNoopenerAndNoreferrerAnchors) {
+                Node targetAttribute = ele.getAttributes().getNamedItem("target");
+                if (targetAttribute != null && targetAttribute.getNodeValue().equalsIgnoreCase("_blank")) {
+                    addNoopenerAndNoreferrer = true;
+                }
+            }
```

Comment:
Combine these two lines too.

Reply:
Do you mean in a single if with `&&`? That would repeat `ele.getAttributes().getNamedItem("target")` for the `null` check and the actual use, seems more readable this way, unless you mean other thing (or disagree).

Reply:
> Do you mean in a single if with `&&`? That would repeat `ele.getAttributes().getNamedItem("target")` for the `null` check and the actual use, seems more readable this way, unless you mean other thing (or disagree).

No - all of these combine line comments refer to having the else line be:   } else {    (all on one line). Simply a formatting comment, not code change.

Reply:
Oh I get it now. This will be the only `else` left then.

Reply:
And it's gone now